frontend/src/metabase/dashboard/reducers.js

```diff
@@ -294,6 +300,32 @@ const parameterValues = handleActions(
   {},
 );
 
+const draftParameterValues = handleActions(
+  {
+    [INITIALIZE]: { next: _state => null }, // reset values
+    [SET_PARAMETER_VALUE]: {
+      next: (state, { payload: { id, value, isDraft } }) => {
+        if (isDraft) {
+          return assoc(state ?? {}, id, value);
+        }
+
+        return state;
+      },
+    },
+    [RESET]: { next: _state => null },
+    [Dashboards.actionTypes.UPDATE]: {
+      next: (state, { payload }) => {
+        if (payload.dashboard.auto_apply_filters) {
+          return null;
```

Comment:
Currently, there's a bug that whenever we update a dashboard, the `parameterValues` will get reset. After that's fixed. I will modify this reducer so that if we update the filter while  auto-apply filters is disabled and toggle the auto-apply filters to true. We should copy the value inside `draftParameterValues` over to `parameterValues` so users don't lose their modified filters.

But to avoid confusion at the moment, and make things simple, I just clear the `draftParameterValues` so the behavior matches what it is at the moment (that the filters will be clear when the dashboard is updated).